sharktank/sharktank/ops/default_impls.py

```diff
@@ -575,12 +575,11 @@ def index_copy__default(
     index = unbox_tensor(index)
     tensor = unbox_tensor(tensor)
     inout_as_torch = unbox_tensor(inout)
-    if (
-        not torch.compiler.is_compiling()
-        and inout_as_torch.is_cpu
-        and inout_as_torch.dtype in [torch.float8_e4m3fnuz, torch.float8_e4m3fn]
-    ):
-        # PyTorch does not have eager implementation for float8_e4m3fnuz in CPU.
+    if not torch.compiler.is_compiling() and inout_as_torch.dtype in [
+        torch.float8_e4m3fnuz,
+        torch.float8_e4m3fn,
+    ]:
+        # PyTorch/PyTorch ROCm does not have eager implementation for various dtypes for CPU/GPU.
```

Comment:
When is this failing?

Also this should ideally be in a separate PR.

Reply:
If you run in eager on GPU. May not have `index_copy_` for these dtypes. I encountered it.

Reply:
Here is a separate PR https://github.com/nod-ai/shark-ai/pull/2389.